lib_eio/unix/timed_semaphore.ml

```diff
@@ -0,0 +1,53 @@
+(* This file is mostly taken from OCaml PR#12867.
+ * It exposes pthread_cond_timedwait(3) for timed_semaphore.ml.
+ * This file (and thread_pool.c) can both be deleted once
+ * this feature is available in OCaml itself. *)
```

Comment:
OCaml is LGPL whereas Eio is ISC, and the OCaml devs have previously not been willing to relicense code for us.

Reply:
(discussed in the dev meeting) The waiting will be done using the scheduler instead, the C code will go away.